doc/contrib/new_api/byref/builders/public.h

```diff
@@ -0,0 +1,69 @@
+#ifndef ELEKTRA_BUILDERS_PUBLIC_H
+#define ELEKTRA_BUILDERS_PUBLIC_H
+
+#include "../core/private.h"
+#include "../core/public.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#define ELEKTRA_KEYSET(...)                                                                                                                \
+	(elektraSetBuild ((ElektraEntry *[]){ __VA_ARGS__ }, sizeof ((ElektraEntry *[]){ __VA_ARGS__ }) / sizeof (ElektraEntry *)))
+
+#define ELEKTRA_KEYNAME(ns_, name_)                                                                                                        \
+	(ElektraName)                                                                                                                      \
+	{                                                                                                                                  \
+		.ns = (ns_), .name = (name_), .size = sizeof (name_)                                                                       \
```

Comment:
That `sizeof` will break as soon as someone passes a dynamically allocated string for `name_`.

Reply:
I know, one solution would be
```suggestion
		.ns = (ns_), .name = (name_), .size = sizeof ("" name_)                                                                       \
```

But the error messages wouldn't be particularly nice. It would probably be something like `expected ')'` and then list the macro expansion.

I just didn't want to clutter this proposal too much with obscure macro tricks ;)